include/zephyr/drivers/fuel_gauge.h

```diff
@@ -37,61 +38,70 @@ extern "C" {
-#define FUEL_GAUGE_STATE_OF_CHARGE	    FUEL_GAUGE_SBS_MFR_ACCESS + 1
+#define FUEL_GAUGE_STATE_OF_CHARGE          FUEL_GAUGE_SBS_MFR_ACCESS + 1
 /** Temperature in 0.1 K */
-#define FUEL_GAUGE_TEMPERATURE		    FUEL_GAUGE_STATE_OF_CHARGE + 1
+#define FUEL_GAUGE_TEMPERATURE              FUEL_GAUGE_STATE_OF_CHARGE + 1
 /** Battery voltage (uV) */
-#define FUEL_GAUGE_VOLTAGE		    FUEL_GAUGE_TEMPERATURE + 1
+#define FUEL_GAUGE_VOLTAGE                  FUEL_GAUGE_TEMPERATURE + 1
 /** Battery Mode (flags) */
-#define FUEL_GAUGE_SBS_MODE		    FUEL_GAUGE_VOLTAGE + 1
+#define FUEL_GAUGE_SBS_MODE                 FUEL_GAUGE_VOLTAGE + 1
 /** Battery desired Max Charging Current (mA) */
-#define FUEL_GAUGE_CHARGE_CURRENT	    FUEL_GAUGE_SBS_MODE + 1
+#define FUEL_GAUGE_CHARGE_CURRENT           FUEL_GAUGE_SBS_MODE + 1
 /** Battery desired Max Charging Voltage (mV) */
-#define FUEL_GAUGE_CHARGE_VOLTAGE	    FUEL_GAUGE_CHARGE_CURRENT + 1
+#define FUEL_GAUGE_CHARGE_VOLTAGE           FUEL_GAUGE_CHARGE_CURRENT + 1
 /** Alarm, Status and Error codes (flags) */
-#define FUEL_GAUGE_STATUS		    FUEL_GAUGE_CHARGE_VOLTAGE + 1
+#define FUEL_GAUGE_STATUS                   FUEL_GAUGE_CHARGE_VOLTAGE + 1
 /** Design Capacity (mAh or 10mWh) */
-#define FUEL_GAUGE_DESIGN_CAPACITY	    FUEL_GAUGE_STATUS + 1
+#define FUEL_GAUGE_DESIGN_CAPACITY          FUEL_GAUGE_STATUS + 1
 /** Design Voltage (mV) */
-#define FUEL_GAUGE_DESIGN_VOLTAGE	    FUEL_GAUGE_DESIGN_CAPACITY + 1
+#define FUEL_GAUGE_DESIGN_VOLTAGE           FUEL_GAUGE_DESIGN_CAPACITY + 1
 /** AtRate (mA or 10 mW) */
-#define FUEL_GAUGE_SBS_ATRATE		    FUEL_GAUGE_DESIGN_VOLTAGE + 1
+#define FUEL_GAUGE_SBS_ATRATE               FUEL_GAUGE_DESIGN_VOLTAGE + 1
 /** AtRateTimeToFull (minutes) */
 #define FUEL_GAUGE_SBS_ATRATE_TIME_TO_FULL  FUEL_GAUGE_SBS_ATRATE + 1
 /** AtRateTimeToEmpty (minutes) */
 #define FUEL_GAUGE_SBS_ATRATE_TIME_TO_EMPTY FUEL_GAUGE_SBS_ATRATE_TIME_TO_FULL + 1
 /** AtRateOK (boolean) */
-#define FUEL_GAUGE_SBS_ATRATE_OK	    FUEL_GAUGE_SBS_ATRATE_TIME_TO_EMPTY + 1
-
+#define FUEL_GAUGE_SBS_ATRATE_OK            FUEL_GAUGE_SBS_ATRATE_TIME_TO_EMPTY + 1
+/** Manufacturing date */
+#define FUEL_GAUGE_MANUFACTURER_DATE        FUEL_GAUGE_SBS_ATRATE_OK + 1
+/** Manufacturer of pack (1 byte length + 20 bytes data) */
+#define FUEL_GAUGE_SERIAL_NUMBER            FUEL_GAUGE_MANUFACTURER_DATE + 1
+/** Manufacturer of pack (1 byte length + 20 bytes data) */
+#define FUEL_GAUGE_MANUFACTURER_NAME        FUEL_GAUGE_SERIAL_NUMBER + 1
+/** Name of pack (1 byte length + 20 bytes data) */
+#define FUEL_GAUGE_DEVICE_NAME              FUEL_GAUGE_MANUFACTURER_NAME + 1
+/** Chemistry (1 byte length + 4 bytes data) */
+#define FUEL_GAUGE_DEVICE_CHEMISTRY         FUEL_GAUGE_DEVICE_NAME + 1
 /** Reserved to demark end of common fuel gauge properties */
-#define FUEL_GAUGE_COMMON_COUNT FUEL_GAUGE_DESIGN_VOLTAGE + 1
+#define FUEL_GAUGE_COMMON_COUNT             FUEL_GAUGE_DEVICE_CHEMISTRY + 1
```

Comment:
How do you feel about us adding a separate range of numbers just for these types of properties? That way when a client uses the wrong kind with the wrong API, it's really easy for the driver to tell the client.

Reply:
@protterc Friendly ping for your comment here.

Reply:
@protterc 
> I like the idea. How about, since it will also affect the register properties, I create a different PR to address - leaving this one a little cleaner? 

Different PR sounds good to me!

> What's your thoughts on COUNT?

Let's make it a MAX with plenty of room for custom properties for downstream users.

> While we're on the subject, any reason these can't/shouldn't be an enum? Something like (would add the comments into the actual enum):
See conversation at https://github.com/zephyrproject-rtos/zephyr/pull/56666#discussion_r1189580654

We had our reasons for this representation (see the linked comment thread) but missed a workaround using `typedef` (I generally avoid `typedef`s). I was going to wait for you to merge this PR before converting things to an `enum` so you wouldn't have to deal with that merge conflict. You're welcome to do the conversion yourself in a different PR too, but I don't mind taking that one since I missed using an `enum` in the first place :)